FaunaDB.Client.Test/EncoderTest.cs

```diff
@@ -352,5 +352,33 @@ public void TestDateTimeOverride()
                 Encode(new DateTimeOverride(testDateTime, testDateTime))
             );
         }
+
+        class StringOverride
+        {
+            [FaunaField("uri")]
+            [FaunaString]
```

Comment:
Can you add other tests ensuring it works on parameters as well (either on constructor or static method)

Reply:
I'm not sure exactly what you mean. Can you give me an example?

Reply:
https://github.com/fauna/faunadb-csharp/blob/master/FaunaDB.Client.Test/DecoderTest.cs#L194
https://github.com/fauna/faunadb-csharp/blob/master/FaunaDB.Client.Test/DecoderTest.cs#L339

Reply:
Added a test and made it work with that

Reply:
also upgraded it to work with primitive types and not just objs with string constructors